packages/exchange/src/pairs/XDaiBridge.ts

```diff
@@ -1,39 +1,126 @@
-const toDaiBridge = '0x7301cfa0e1756b71869e93d4e4dca5c7d0eb0aa6';
-
-export default class XDaiBridge extends Pair {
-  constructor() {
-    super({ assetA: 'xdai', assetB: 'dai' });
+const bridgeAAbi: AbiItem[] = [
+  {
+    "anonymous": false,
+    "inputs": [
+      {
+        "indexed": false,
+        "name": "recipient",
+        "type": "address"
+      },
+      {
+        "indexed": false,
+        "name": "value",
+        "type": "uint256"
+      },
+      {
+        "indexed": false,
+        "name": "transactionHash",
+        "type": "bytes32"
+      }
+    ],
+    "name": "AffirmationCompleted",
+    "type": "event"
   }
+]
 
-  exchangeAtoB({ account, value, ether }: ExchangeParams) {
-    const _value = this._getValue({ value, ether });
-    const xdai = this.getExchange().getAsset('xdai');
-    return xdai.send({
-      from: account,
-      value: _value,
-      to: toDaiBridge,
-    });
+const bridgeBAbi: AbiItem[] = [
+  {
+    "anonymous": false,
+    "inputs": [
+      {
+        "indexed": false,
+        "name": "recipient",
+        "type": "address"
+      },
+      {
+        "indexed": false,
+        "name": "value",
+        "type": "uint256"
+      },
+      {
+        "indexed": false,
+        "name": "transactionHash",
+        "type": "bytes32"
+      }
+    ],
+    "name": "RelayedMessage",
+    "type": "event"
   }
+]
 
-  exchangeBtoA({ account, value, ether }: ExchangeParams) {
-    const _value = this._getValue({ value, ether });
+const wait = (time: number) => new Promise(resolve => setTimeout(resolve, time));
 
-    const dai = this.getExchange().getAsset(this.assetB);
-    return dai.send({
-      from: account,
-      value: _value,
-      to: toXdaiBridge,
+export default class XDaiBridge extends Bridge {
+  constructor() {
+    super({
+      assetA: 'xdai',
+      assetABridge: '0x7301cfa0e1756b71869e93d4e4dca5c7d0eb0aa6',
+      assetB: 'dai',
+      assetBBridge: '0x4aa42145Aa6Ebf72e164C9bBC74fbD3788045016'
     });
   }
 
-  async estimateAtoB(value: ValueTypes) {
-    return this._getValue(value);
+  async detectExchangeBToAFinished(account: string, value: string, sendResult: any) {
+    const asset = this.getExchange().getAsset(this.assetA);
+    const web3 = asset.getWeb3();
+    const contract = new web3.eth.Contract(bridgeAAbi, this.assetABridge);
+    let fromBlock = await web3.eth.getBlockNumber()
+
+    while (true) {
+      const currentBlock = await web3.eth.getBlockNumber()
+
+      const events: EventData[] = await contract.getPastEvents('AffirmationCompleted', {
+        fromBlock,
+        toBlock: currentBlock
+      })
+
+      const confirmationEvent = events.filter(event => event.returnValues.transactionHash === sendResult.txHash)
+
+      if (confirmationEvent.length > 0) {
+        const event = confirmationEvent[0]
+        const block = await web3.eth.getBlock(event.blockNumber);
+
+        const historyEvent = {
```

Comment:
Don't you think that `historyEvent` construction and and adding it to the history should be implemented within `scanBlocks` of `NativeAssets` (https://github.com/austintgriffith/burner-core/blob/de48a79037c8d3cba6aa13ea0a11a78d8677e34b/assets/src/NativeAsset.js#L45)?

Reply:
Yes, it looks like it is the correct place to add events to history.

I think I could remove creating the event from `XDaiBridge` and open another PR in `burner-core` to extend the `scanBlocks` method in the [xdai](https://github.com/austintgriffith/burner-core/blob/de48a79037c8d3cba6aa13ea0a11a78d8677e34b/assets/src/xdai.js)  asset to also look for the `AffirmationCompleted` events in the bridge. Does it look good to you?

Reply:
Yes, it works for me. As part of the changes for `burner-core` and in order to reduce number of PRs, please consider to introduce the changes there that could be used by AMB as well.

Reply:
The creation of the event was removed https://github.com/burner-wallet/burner-wallet-2/pull/21/commits/44c9510bce2700290d8ce49859febf67642e7ecf